frontend/src/app/libs/common-components/components/input/input.component.ts

```diff
@@ -0,0 +1,85 @@
+import { ThrowStmt } from '@angular/compiler';
+import { AfterViewInit, ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { ComponentTheme } from '../../shared/component-theme.enum';
+import { InputType } from '../../shared/input-type.enum';
+
+@Component({
+	selector: 'app-input',
+	templateUrl: './input.component.html',
+	styleUrls: ['./input.component.scss'],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InputComponent {
+
+	@Input()
+	public hide: boolean = true;
+
+	@Input()
+	public inputType: string;
+
+	@Input()
+	public label: string = 'label';
+
+	@Input()
+	public placeholder: string = 'placeholder';
+
+	@Input()
+	public isRequired: boolean = true;
+
+	@Input()
+	public theme: ComponentTheme = ComponentTheme.Light;
+
+	@Input()
+	public set defaultValue(value: string) {
+		this.formControlName.setValue(value);
+	}
+
+	public get isDarkTheme(): boolean {
+		return this.theme === ComponentTheme.Dark;
+	}
+
+	public textFormControl: FormControl = new FormControl('', [
```

Comment:
no need to create 3 instance of form control each time, just choose one based on `[type]`